src/hotspot/share/runtime/fieldDescriptor.cpp

```diff
@@ -216,18 +207,33 @@ void fieldDescriptor::print_on_for(outputStream* st, oop obj) {
       ShouldNotReachHere();
       break;
   }
-  // Print a hint as to the underlying integer representation. This can be wrong for
-  // pointers on an LP64 machine
+
+  // Print  a hint as to the underlying integer representation.
+  if (is_reference_type(ft)) {
 #ifdef _LP64
-  if (is_reference_type(ft) && UseCompressedOops) {
-    st->print(" (%x)", obj->int_field(offset()));
-  }
-  else // <- intended
+    if (UseCompressedOops) {
+      st->print(" (" PTR32_FORMAT ")", obj->int_field(offset()));
+    } else {
+      st->print(" (" PTR_FORMAT ")", obj->long_field(offset()));
+    }
+#else
+    st->print(" (" PTR32_FORMAT ")", obj->int_field(offset()));
 #endif
-  if (ft == T_LONG || ft == T_DOUBLE LP64_ONLY(|| !is_java_primitive(ft)) ) {
-    st->print(" (%x %x)", obj->int_field(offset()), obj->int_field(offset()+sizeof(jint)));
-  } else if (as_int < 0 || as_int > 9) {
-    st->print(" (%x)", as_int);
+  } else { // Primitives
+    if (ft == T_LONG || ft == T_DOUBLE) {
+      st->print(" (" PTR64_FORMAT ")", obj->long_field(offset()));
+    } else {
+      switch (ft) {
+        case T_BYTE:    st->print(" (" PTR8_FORMAT  ")", obj->byte_field(offset()));  break;
```

Comment:
Suggestion: INT8_FORMAT_X as this is not a PTR

Ditto for 16, 32, and 64.

Reply:
I had the same thought first. However, if I use the INT/UINT versions that the output will be: `80041dc0` instead of `0x0000000080041dc0`. I prefer the latter, and that is probably a personal taste, but it also makes it a little less tedious to use in a debugger if 0x is already prepended.

Reply:
Maybe we can find a better name, that still prints the 0x00<...> format? I agree that these PTR32, PTR16, PTR8 is oddly named. Let me think about alternative names.